### PR TITLE
Introduce tests for rule types

### PR DESCRIPTION
**Pull request by JAORMX**
This uses Minder's testkit in order to create tests for rule types. A
sample is given for the `license` and `secret_scanning_enabled` rule
types.

Signed-off-by: Juan Antonio Osorio <ozz@stacklok.com>

### CODE REVIEW

File: rule-types/github/secret_scanning.test.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        owner: "coolhead"
+        name: "haze-wave"
+    http:
+      body_file: enabled.json
+  - name: "Should have secret scanning enabled for private repo"
+    def:
+      skip_private_repos: false
+    params: {}
+    expect: "pass"
+    entity: *test-repo
```
Comment (jhrozek) [inline rule-types/github/secret_scanning.test.yaml:18]: Does this refer to the test-repo defined above?

Comment (JAORMX) [reply rule-types/github/secret_scanning.test.yaml:18]: that's right, I'm using a YAML anchor here.

File: rules_test.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+// The Entity Key in EntityVersionWrapper requires a custom unmarshaler
+// Version and Type can be parsed as is.
+// The Entity field depends on the Type field.
+func (e *EntityVersionWrapper) UnmarshalYAML(value *yaml.Node) error {
+	var entity map[string]any
+	if err := value.Decode(&entity); err != nil {
+		return err
+	}
+
+	e.Type = entity["type"].(string)
```
Comment (jhrozek) [inline rules_test.go:59]: If type is missing or not a string, then this would panic right?

**Review by jhrozek — DISMISSED** [review]
The tests seem to run and modifying the test data causes failures. I left a nit inline, but since this is test code, feel free to go ahead and merge.

**Review by jhrozek — APPROVED** [review]

**Pull request merged**